src/molecules/service.py

```diff
@@ -169,8 +181,7 @@ def process_csv_file(self, file: UploadFile):
         :return: Number of molecules added successfully
         """
 
-        contents = file.file.read().decode("utf-8")
-        csv_reader = csv.DictReader(contents.splitlines())
+        csv_reader = csv.DictReader(io.TextIOWrapper(file.file, encoding="utf-8"))
```

Comment:
I also optimized file reading here, instead of reading the entire file with  file.file.read().decode("utf-8"), I used io.TextIOWrapper(file.file, encoding="utf-8")